src/Query/Filter/TextField.ts

```diff
@@ -44,17 +43,24 @@ export abstract class TextField extends Field {
         // Finally, we can create the Filter, that takes a task
         // and tests if it matches the string filtering rule
         // represented by this object.
-        return FilterOrErrorMessage.fromFilter((task: Task) => {
-            return TextField.maybeNegate(matcher!.matches(this.value(task)), filterMethod);
-        });
+        const negate = operator.match(/not/) !== null;
+        return FilterOrErrorMessage.fromFilter(this.getFilter(matcher, negate));
     }
 
     public static stringIncludesCaseInsensitive(haystack: string, needle: string): boolean {
         return SubstringMatcher.stringIncludesCaseInsensitive(haystack, needle);
     }
 
+    protected fieldPattern(): string {
+        return this.fieldName();
+    }
+
+    protected operatorPattern(): string {
```

Comment:
As with `fieldPattern`, please add JSDoc and say what is meant by operator.

Reply:
Improved this as well.